### PR TITLE
Issues/19 add detail page

### PR DESCRIPTION
**Pull request by azar-barfipour**
I added detail page for each job, when click on the job, the page shows. For now except title, other data are dummy.

### CODE REVIEW

File: src/pages/job-detail.js [hunk 1]
```diff
@@ -0,0 +1,31 @@
+import JobDetailItem from "../components/organisms/job-detail/job-detail-item";
+import { useEffect, useState } from "react";
+import { useParams } from "react-router-dom";
+
+const JobDetail = () => {
+  const [jobDetailList, setJobDetailList] = useState([]);
```
Comment (SOunit) [inline src/pages/job-detail.js:6]: job detail page has single job right?
so `[jobDetail, setJobDetail]` may be better.

File: src/pages/job-detail.js [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+  const params = useParams();
+  useEffect(() => {
+    const fetchJobDetails = async () => {
```
Comment (SOunit) [inline src/pages/job-detail.js:9]: same.
job detail should be single.

File: src/pages/job-detail.js [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+      try {
+        const response = await fetch(
+          `${process.env.REACT_APP_BACKEND_URL}/jobs/${params.jobId}`
+        );
+        const data = await response.json();
+        console.log(data.job);
```
Comment (SOunit) [inline src/pages/job-detail.js:15]: please delete log.

File: src/pages/job-detail.js [hunk 4]
```diff
@@ -0,0 +1,31 @@
...
+
+        setJobDetailList(data.job);
+      } catch (err) {
+        console.log(err.message);
+      }
+    };
+    fetchJobDetails();
+  }, []);
+
+  return <JobDetailItem jobDetailList={jobDetailList} />;
```
Comment (SOunit) [inline src/pages/job-detail.js:29]: prop could be `jobDetail`

File: src/components/organisms/job-detail/job-detail-item.js [hunk 5]
```diff
@@ -0,0 +1,214 @@
+import Button from "../../atoms/button";
+import classes from "./job-detail-item.module.css";
+import SkillList from "./job-detail-skill";
+
+const DUMMIDATA = [
```
Comment (SOunit) [inline src/components/organisms/job-detail/job-detail-item.js:5]: variable name `DUMMY_DATA` is better?

File: src/components/organisms/job-detail/job-detail-item.js [hunk 6]
```diff
@@ -0,0 +1,214 @@
...
+    postedAt: "1mo ago",
+    contract: "Part Time",
+    location: "Worldwide",
+    languages: ["JavaScript"],
+    tools: ["React", "Sass"],
+  },
+];
+
+const JobDetailItem = (props) => {
+  return (
```
Comment (SOunit) [inline src/components/organisms/job-detail/job-detail-item.js:159]: may i ask you to spread props?
`(props)` => `({jobDataList})`

Comment (SOunit) [reply src/components/organisms/job-detail/job-detail-item.js:159]: props name could be `jobDetail`

File: src/components/organisms/job-detail/job-detail-item.js (hunk 5 above)
Comment (SOunit) [reply src/components/organisms/job-detail/job-detail-item.js:5]: let's create `dummy-data.js` in `src/components/test` and import the data from outer file

File: src/components/organisms/job-detail/job-detail-item.js [hunk 7]
```diff
@@ -0,0 +1,214 @@
...
+    <li className={classes["jobDetail-item"]}>
```
Comment (SOunit) [inline src/components/organisms/job-detail/job-detail-item.js:160]: its single item so i guess we can delete li tag

File: src/components/organisms/job-list/job-card.js [hunk 8]
```diff
@@ -1,16 +1,22 @@
...
 const JobCard = (props) => {
   return (
     <Card className={`${classes[componentName]} ${props.className}`}>
-      <h2 className={`${classes[componentName]}__title`}>{props.job.title}</h2>
-      <div>
-        {props.job.skills.map((skill) => (
-          <div key={skill.id}>{skill.title}</div>
-        ))}
-      </div>
+      <Link
```
Comment (SOunit) [inline src/components/organisms/job-list/job-card.js:9]: is it possible to wrap card with link?
so that all card is touchable.

Comment (SOunit) [reply src/components/organisms/job-list/job-card.js:9]:
```
<Link
        to={`/jobs/detail/${props.job.id}`}
        className={classes[`${componentName}-link`]}
      >
```

**Review by SOunit — CHANGES_REQUESTED** [review]
thanks for contribution!

I added some comments to fix!

File: src/components/organisms/job-detail/job-detail-item.js (hunk 5 above)
Comment (azar-barfipour) [reply src/components/organisms/job-detail/job-detail-item.js:5]: Aren't we use actual data soon ?

File: src/components/organisms/job-list/job-card.js (hunk 8 above)
Comment (azar-barfipour) [reply src/components/organisms/job-list/job-card.js:9]: I did that but it makes the like button link too. which is not correct

File: src/components/organisms/job-detail/job-detail-item.js (hunk 5 above)
Comment (SOunit) [reply src/components/organisms/job-detail/job-detail-item.js:6]: sorry i have no time to setup another data, so let's use dummy data for now!

File: src/components/organisms/job-list/job-card.js (hunk 8 above)
Comment (SOunit) [reply src/components/organisms/job-list/job-card.js:9]: ok, then let's keep it as it is!

**Review by SOunit — APPROVED** [review]

**Review by SOunit — APPROVED** [review]

**Pull request merged**